src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -510,6 +538,17 @@ def _on_filename(self, name):
         """
         called when the .filename VA changes
         """
+        # For the first feature acquisition, reset the counter to zero
+        if self.overview_acqui_controller._tab_data_model.main.currentFeature.value is None:
+            self._config.fn_count = '0'
+            self._filename.value = create_filename(
+                self._config.pj_last_path,
+                self._config.fn_ptn,
+                self._config.last_extension,
+                self._config.fn_count,
+            )
+            name = self._filename.value
+
```

Comment:
I don't think you need that. `create_filename()` will automatically look at the current files and increase the counter if needed.